tests/test_api.py

```diff
@@ -170,17 +173,12 @@ def test_insecure_connection(self):
         call request with verify=False.
         """
         with mock.patch('requests.sessions.Session.request') as g:
-            mock_response = type('statobj', (), {})()  # placeholder object
-            mock_response.status_code = 200
-            g.return_value = mock_response
-            with client.test_mode as t:
-                t.register('/ping/', "I'm a teapot!", status_code=200)
+            with client.test_mode:
                 with settings.runtime_values(verify_ssl=False):
-                    client.get('/ping/')
+                    client._make_request('GET', '/api/', [], {})
```

Comment:
This patching was pretty complicated beforehand - it was relying on some very specific implementation details of the requests library to simulate fake response objects.  It was also hard to mock properly because now there's two requests being made (one to determine the API version, and the other for `/api/v1/ping/`.

So I've simplified this test to really just perform the API version detection request so that we can make sure that `verify=False` was used properly (which is the whole point of this test).